airflow/models/dagrun.py

```diff
@@ -207,6 +207,20 @@ def refresh_from_db(self, session: Session = None):
         self.id = dr.id
         self.state = dr.state
 
+    @classmethod
+    @provide_session
+    def active_runs_of_dags(cls, dag_ids=None, only_running=False, session=None) -> Dict[str, int]:
+        """Get the number of active dag runs for each dag."""
+        query = session.query(cls.dag_id, func.count('*'))
+        if dag_ids:
+            query = query.filter(cls.dag_id.in_(list(set(dag_ids))))
```

Comment:
Worthwhile to move the comment on deduplicating here. Also, if `dag_ids` is an empty list, I feel we should filter against an empty list instead of not filtering.

```suggestion
        if dag_ids is not None:
            # 'set' called to avoid duplicate dag_ids, but converted back to 'list'
            # because SQLAlchemy doesn't accept a set here.
            query = query.filter(cls.dag_id.in_(list(set(dag_ids))))
```